src/cozmo/world.py

```diff
@@ -611,13 +611,64 @@ def send_available_objects(self):
         self.conn.send_msg(msg)
 
+    def _remove_custom_marker_object_instances(self):
+        for id, obj in list(self._objects.items()):
+            if isinstance(obj, objects.CustomObject):
+                logger.info("Removing CustomObject instance: id %s = obj '%s'", id, obj)
+                del self._objects[id]
+
+    def _remove_fixed_custom_object_instances(self):
+        for id, obj in list(self._objects.items()):
+            if isinstance(obj, objects.FixedCustomObject):
+                logger.info("Removing FixedCustomObject instance: id %s = obj '%s'", id, obj)
+                del self._objects[id]
+
     async def delete_all_custom_objects(self):
-        """Causes the robot to forget about all custom objects it currently knows about."""
+        """Causes the robot to forget about all custom (fixed + marker) objects it currently knows about.
+        
+        Note: This includes all fixed objects, and all custom marker object instances, 
+        BUT this does NOT remove the custom marker object definitions, so Cozmo
+        will continue to add new objects if he sees the markers again. To remove
+        the definitions for those objects use: :meth:`undefine_all_custom_marker_objects`
+        """
         msg = _clad_to_engine_iface.DeleteAllCustomObjects()
         self.conn.send_msg(msg)
-        # TODO: use a filter to wait only for a message for the active robot
         await self.wait_for(_clad._MsgRobotDeletedAllCustomObjects)
-        # TODO: reset local object stte
+        self._remove_custom_marker_object_instances()
+        self._remove_fixed_custom_object_instances()
+
+    async def delete_custom_marker_objects(self):
+        """Causes the robot to forget about all custom marker objects it currently knows about.
+
+        Note: This removes custom marker object instances only, it does NOT remove
+        fixed objects, nor does it remove the custom marker object definitions, so Cozmo
+        will continue to add new objects if he sees the markers again. To remove
+        the definitions for those objects use: :meth:`undefine_all_custom_marker_objects`
+        """
+        msg = _clad_to_engine_iface.DeleteCustomMarkerObjects()
+        self.conn.send_msg(msg)
+        await self.wait_for(_clad._MsgRobotDeletedCustomMarkerObjects)
+        self._remove_custom_marker_object_instances()
+
+    async def delete_fixed_custom_objects(self):
+        """Causes the robot to forget about all fixed custom objects it currently knows about.
+
+        Note: This removes fixed custom objects only, it does NOT remove
+        the custom marker object instances or definitions.
+        """
+        msg = _clad_to_engine_iface.DeleteFixedCustomObjects()
+        self.conn.send_msg(msg)
+        await self.wait_for(_clad._MsgRobotDeletedFixedCustomObjects)
+        self._remove_fixed_custom_object_instances()
+
+    async def undefine_all_custom_marker_objects(self):
```

Comment:
No, they're not defined - the instances are added to the world but there's no equivalent of the custom marker object definitions (which state e.g. there's a box X,Y,Z with these markers on it).